pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -72,3 +83,66 @@ func FilterFunctionTags(input map[string]string) map[string]string {
 
 	return output
 }
+
+func CreateInferredSpan(event string, inferredSpan InferredSpan) {
+	// Parse the event into the EventKey struct
+	eventSource, attributes := ParseEventSource(event)
+	switch eventSource {
+	case "apigateway":
+		CreateInferredSpanFromAPIGatewayEvent(eventSource, attributes, inferredSpan)
+	case "http-api":
+		CreateInferredSpanFromAPIGatewayHTTPEvent(eventSource, attributes, inferredSpan)
+	case "websocket":
+		CreateInferredSpanFromAPIGatewayWebsocketEvent(eventSource, attributes, inferredSpan)
+	}
+}
+
+func CompleteInferredSpan(
+	processTrace func(p *api.Payload),
+	endTime time.Time,
+	isError bool,
+	requestId string,
+	inferredSpan InferredSpan) {
+
+	if inferredSpan.IsAsync {
+		inferredSpan.Span.Duration = inferredSpan.Span.Start
+	} else {
+		inferredSpan.Span.Duration = endTime.UnixNano() - inferredSpan.Span.Start
+	}
+	if isError {
+		inferredSpan.Span.Error = 1
+	}
+	traceChunk := &pb.TraceChunk{
+		Priority: int32(sampler.PriorityNone),
+		Origin:   "lambda",
+		Spans:    []*pb.Span{inferredSpan.Span},
+	}
+
+	tracerPayload := &pb.TracerPayload{
+		Chunks: []*pb.TraceChunk{traceChunk},
+	}
+
+	processTrace(&api.Payload{
+		Source:        info.NewReceiverStats().GetTagStats(info.Tags{}),
+		TracerPayload: tracerPayload,
+	})
+}
+
+// GenerateSpan declares and initializes a new inferred span
+// with the SpanID and TraceID
+func GenerateSpan() InferredSpan {
```

Comment:
Nit: I would rename this to `GenerateInferredSpan` because its return type is `InferredSpan` (not `Span`)